Source/WebKit/UIProcess/ProvisionalPageProxy.cpp

```diff
@@ -170,6 +170,23 @@ void ProvisionalPageProxy::initializeWebPage(RefPtr<API::WebsitePolicies>&& webs
 
     auto parameters = m_page->creationParameters(m_process, *m_drawingArea, WTFMove(websitePolicies));
     parameters.isProcessSwap = true;
+
+    auto* openerFrame = m_page->openerFrame();
+    if (openerFrame) {
+        if (auto* openerPage = openerFrame->page()) {
+            parameters.openerFrameIdentifier = openerFrame->frameID();
+            auto parameters = openerPage->creationParameters(m_process, *openerPage->drawingArea(), openerPage->lastWebsitePolicies());
+
+            parameters.subframeProcessFrameTreeInitializationParameters = { {
+                openerFrame->frameID(),
+                *openerPage->frameTreeCreationParameters(),
+                std::nullopt,
+            } };
+
+            m_process->send(Messages::WebProcess::CreateWebPage(openerPage->webPageID(), parameters), 0);
```

Comment:
A RemoteFrame requires a Page and in this situation it wouldn't be correct to claim that the opened page is the associated page for the frame.

Reply:
Is there a UIProcess object that represents this page?

Reply:
If we really need this page in the process (Apparently Alex didn't say otherwise), and if we need a UIProcess-side object to represent it, I recommend we use the same object that Alex already introduced for remote pages (Please let's not add an OpenerPageProxy like the earlier patch iteration). I don't remember what it is called.